### ISSUE TITLE
Could not register the vSphere extension due to err: context deadline exceeded 

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#1966]
Failed to deploy VIC to a nimbus created VCH recently with this error:
```
[34mINFO[0m[2016-08-13T07:32:30-05:00] Uploading images for container [34mINFO[0m[2016-08-13T07:32:30-05:00] "bin/bootstrap.iso" [34mINFO[0m[2016-08-13T07:32:30-05:00] "bin/appliance.iso" [34mINFO[0m[2016-08-13T07:43:24-05:00] Registering VCH as a vSphere extension [31mERRO[0m[2016-08-13T07:43:24-05:00] Could not register the vSphere extension due to err: context deadline exceeded [34mINFO[0m[2016-08-13T07:43:24-05:00] Collecting 740d1b11-ac09-454a-ac12-9e9b4887f805 vpxd.log [31mERRO[0m[2016-08-13T07:43:24-05:00] Failed to collect 740d1b11-ac09-454a-ac12-9e9b4887f805 vpxd.log: context deadline exceeded [33mWARN[0m[2016-08-13T07:43:24-05:00] No log data for 740d1b11-ac09-454a-ac12-9e9b4887f805 vpxd.log [31mERRO[0m[2016-08-13T07:43:24-05:00] -------------------- [31mERRO[0m[2016-08-13T07:43:24-05:00] vic-machine-linux failed: Error registering VCH vSphere extension: context deadline exceeded
```

[vmware.zip](https://github.com/vmware/vic/files/416741/vmware.zip)

**Comment by emlin** [comment]
@mhagen-vmware can I access your vc env? I didn't see obviously error from vpxd log

**Comment by mhagen-vmware** [comment]
env doesn't exist anymore.  And this passed once I was on the office network this morning.

**Comment by mhagen-vmware** [comment]
If that error is coming from a normal VC operation, maybe we can have lifecycle team take a look at the bundle?

**Comment by emlin** [comment]
Yeah, that is a normal VC operation, to register one extension only. It would be better to have lifecycle team have a look, or file a bug to VC team.

**Comment by emlin** [comment]
From our side, not much to do, to be honest.

**Comment by emlin** [comment]
@mhagen-vmware are you still seeing this?

**Comment by mhagen-vmware** [comment]
Haven't tried this at home again since.

**Comment by emlin** [comment]
@mhagen-vmware The timeout is for almost the whole create operation. From your vic-machine log, the image upload takes 11m, and after that extension registration failed immediately.
You probably set timeout to 15m, including resource validation, there is no time left for anything else. 
Please try longer timeout to see if anything different.

**Comment by mhagen-vmware** [comment]
I set the timeout to 30min and 60min as well and saw the same error.

**Comment by emlin** [comment]
Then please post that output as well. From this one, not likely it is 30 or 60 minutes.

I think this is still because of your slow env, cause I cannot reproduce it through wifi and vpn. But the output could help me to understand what's happening.

**Comment by mhagen-vmware** [comment]
But if it took 11 minutes to upload and we still had 4 minutes left, why would it take over 4 minutes to just register the extension which was the next step after upload right? That should never take that long especially on a system that is sitting idle not doing anything else right?

**Comment by emlin** [comment]
The other 4 minutes are just some time might be used before upload images, cause vic-machine will do validation for all configured parameters.
Please post all vic-machine log if next time this issue is reproduced, then we can have clear idea for how long time each step took.

**Comment by derekbeard** [comment]
Hi Mike and Emma. I don't see any failed ExtensionManager.registerExtension calls in that support bundle.  Emma, can you confirm that I'm looking for the correct API?

Thanks,

Derek

**Comment by emlin** [comment]
@derekbeard you're right.

But from the vic-machine output, we can see that extension registration failed immediately after image upload. That means client might not send out extension registration at all.
So this issue might be caused by the slow network connection through VPN. 
@mhagen-vmware see this again?

**Comment by mhagen-vmware** [comment]
Closing as not reproducible. I believe I have figured out what happened here. My laptop doesn't appear to be completely interoperable wirelessly with my home network, so the connection continuously drops. (I have since updated the firmware, hopefully that fixes it) I think what happened here is that the initial upload of the ISOs succeeded but the connection dropped immediately afterwards, in particular my home network will maintain connection during a continuous stream of data, but as soon as there is a break in the stream the connection will drop which is why the uploads probably finished and immediately dropped connection. which hung my VPN connection to the corp network and caused the next operation, registering the extension to fail.
